src/test/netbase_tests.cpp

```diff
@@ -45,6 +45,30 @@ BOOST_AUTO_TEST_CASE(netbase_properties)
     BOOST_CHECK(CNetAddr("8.8.8.8").IsRoutable());
     BOOST_CHECK(CNetAddr("2001::1").IsRoutable());
     BOOST_CHECK(CNetAddr("127.0.0.1").IsValid());
+    BOOST_CHECK(CNetAddr("1.1.1").ToString() == "1.1.0.1");
```

Comment:
> parsing should be more strict and providing a partial IPv4 address should simply fail (return !IsValid()).

Agreed, I want to move on this slowly and carefully to ensure we don't introduce a crash via IP strings which work now but should fail. Should I open an issue for this?